filters/tee/teeloopback_test.go

```diff
@@ -0,0 +1,113 @@
+package tee
+
+import (
+	"fmt"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/predicates/primitive"
+	teePredicate "github.com/zalando/skipper/predicates/tee"
+	"github.com/zalando/skipper/proxy/backendtest"
+	"github.com/zalando/skipper/proxy/proxytest"
+	"github.com/zalando/skipper/routing"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+func waitForAll(channels []backendtest.Done, done chan struct{}) {
+	var wg sync.WaitGroup
```

Comment:
this is possible without many goroutines and without the waitgroup:

```
func waitForAll(channels []backendtest.Done, done chan struct{}) {
	for _, ch := range channels {
		<-ch
	}

	close(done)
}
```

Reply:
Nice!